Makefile

```diff
@@ -431,24 +412,37 @@ define wait
 endef
 
 define generate
-	$(call controller-gen,object:headerFile="hack/boilerplate.go.txt" paths="./...")
+	$(call controller-gen,object:headerFile="hack/boilerplate.gotxt" paths="./...")
 endef
 
 define manifest
 	$(call controller-gen,crd:crdVersions=v1 paths="./..." output:crd:artifacts:config=chart/k8gb/templates/)
 endef
 
-# function retrieves controller-gen path or installs controller-gen@v3.0.0 and retrieve new path in case it is not installed
 define controller-gen
 	@$(if $(filter $(CONTROLLER_GEN_PATH),$(NO_VALUE)),$(call install-controller-gen),)
+	@$(if $(filter-out $(CONTROLLER_GEN_VERSION),$(CONTROLLER_GEN_CURRENT_VERSION)),$(call install-controller-gen),)
 	$(CONTROLLER_GEN_PATH) $1
 endef
 
 define install-controller-gen
-	GO111MODULE=on go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.5.0
+	@echo "\n$(YELLOW)Installing controller-gen $(CYAN)$(CONTROLLER_GEN_VERSION)$(NC)"
+	GO111MODULE=on go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_GEN_VERSION)
 	$(eval CONTROLLER_GEN_PATH = $(GOBIN)/controller-gen)
 endef
 
+define golic
+	@$(if $(filter $(GOLIC_PATH),$(NO_VALUE)),$(call install-golic),)
+	@$(if $(filter-out $(GOLIC_VERSION),$(GOLIC_CURRENT_VERSION)),$(call install-golic),)
+	$(GOLIC_PATH) inject $1
+endef
+
+define install-golic
+	@echo "\n$(YELLOW)Installing golic $(CYAN)$(GOLIC_VERSION)$(NC)"
+	GO111MODULE=on go install github.com/AbsaOSS/golic@$(GOLIC_VERSION)
```

Comment:
Go1.16 [enables  GO111MODULE=on by default](https://medium.com/codex/go-1-16-is-launched-9ae3cf06703a)

_Enabling Go Module by default
We are not required to declare GO111MODULE anymore when this environment variable is enabled by default. It’s 2021 now, and who hasn’t used Go Modules yet?
Go install command
The go install command can install go programs without affecting mod file dependencies. For example, executing go install module1/test@0.0.1 plus the version number is enough. And in the future, go get will no longer be accepted to compile and install, and you can close it with -d option, which means go install will be the only installation command._